src/ConnectionAutomaton.cpp

```diff
@@ -286,7 +286,7 @@ void ConnectionAutomaton::reconnect(
 
 void ConnectionAutomaton::addStream(
     StreamId streamId,
-    std::shared_ptr<StreamAutomatonBase> automaton) {
+    yarpl::Reference<StreamAutomatonBase> automaton) {
```

Comment:
I don't see why we'd change this? Why use our home-made shared_ptr instead of shared_ptr here?